File: Python/models/registration.py
from typing import Any, Final
from datetime import date


class Registration:
    def __init__(self, event_id: int, attendee_id: int, registered_at: date) -> None:
        self._event_id: Final[int] = self._validate_type(
            "event_id", event_id, int)
        self._attendee_id: Final[int] = self._validate_type(
            "attendee_id", attendee_id, int)
        self._registered_at: Final[date] = self._validate_type(
            "registered_at", registered_at, date)

    @property
    def event_id(self) -> int:
        return self._event_id

    @property
    def attendee_id(self) -> int:
        return self._attendee_id

    @property
    def registered_at(self) -> date:
        return self._registered_at

    @staticmethod
    def _validate_type(param_name: str, value: Any, expected_type: type) -> None:
        if not isinstance(value, expected_type):
            raise TypeError(
                f"{param_name} must be a {expected_type}, got {type(value).__name__}")
        return value

    def get_insert_registration_query(self) -> str:
        return f"INSERT INTO registrations (event_id, attendee_id, registered_at) VALUES ('{self.event_id}', '{self.attendee_id}', '{self.registered_at}');"

File: Go/models/registration.go
package models

import (
	"fmt"
	"time"
)

type Registration struct {
	eventID          int
	attendeeID       int
	registrationDate time.Time
}

func NewRegistration(eventID, attendeeID int, registrationDate time.Time) *Registration {
	return &Registration{
		eventID:          eventID,
		attendeeID:       attendeeID,
		registrationDate: registrationDate,
	}
}

func (r *Registration) GetInsertRegistrationQuery() string {
	return fmt.Sprintf("INSERT INTO registrations (event_id, attendee_id, registered_at) VALUES (%d, %d, '%s');", r.eventID, r.attendeeID, r.registrationDate.Format("2006-01-02 15:04:05"))
}
